falcon-regression/merlin-core/src/main/java/org/apache/falcon/regression/core/util/OozieUtil.java

```diff
@@ -714,7 +714,9 @@ public static String getActionStatus(OozieClient oozieClient, String workflowId,
         List<WorkflowAction> wfAction = oozieClient.getJobInfo(workflowId).getActions();
         for (WorkflowAction wf : wfAction) {
             if (wf.getName().contains(actionName)) {
-                return wf.getExternalStatus();
+                if (wf.getExternalStatus() != null) {
```

Comment:
please use 'and' operator in if condition and combine both the if statements for better readability.